pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -152,26 +152,33 @@ type ruleCache struct {
 	podUpdates <-chan v1beta1.PodReference
 }
 
-func (c *ruleCache) getNetworkPolicies(namespace string) []v1beta1.NetworkPolicy {
-	ret := []v1beta1.NetworkPolicy{}
+func (c *ruleCache) getNetworkPolicies(npFilter v1beta1.NetworkPolicyQueryFilter) []v1beta1.NetworkPolicy {
+	var ret []v1beta1.NetworkPolicy
 	c.policyMapLock.RLock()
 	defer c.policyMapLock.RUnlock()
-	for uid, np := range c.policyMap {
-		if namespace == "" || np.Namespace == namespace {
+	for uid, npr := range c.policyMap {
+		if c.networkPolicyMatchFilter(npFilter, npr) {
 			ret = append(ret, *c.buildNetworkPolicyFromRules(uid))
 		}
 	}
 	return ret
 }
 
+// If this npr(Network Policy Reference) can match the npFilter(Network Policy Filter)
+func (c *ruleCache) networkPolicyMatchFilter(npFilter v1beta1.NetworkPolicyQueryFilter, npr v1beta1.NetworkPolicyReference) bool {
+	return (npFilter.Name == "" || npFilter.Name == npr.Name) &&
+		(npFilter.Namespace == "" || npFilter.Namespace == npr.Namespace) &&
+		(npFilter.SourceType == "" || string(npFilter.SourceType) == string(npr.Type))
+}
+
 // getNetworkPolicy looks up and returns the cached NetworkPolicy.
 // nil is returned if the specified NetworkPolicy is not found.
-func (c *ruleCache) getNetworkPolicy(npName, npNamespace string) *v1beta1.NetworkPolicy {
+func (c *ruleCache) getNetworkPolicy(npFilter v1beta1.NetworkPolicyQueryFilter) *v1beta1.NetworkPolicy {
```

Comment:
There are two functions: `getNetworkPolicy` and `getNetworkPolicies`. 
According to my understanding, previously, before supporting CNP and ANP, `getNetworkPolicy` is used to get a single NP with name and namespace. `getNetworkPolicies` is used to get a list of NP in one namespace or all namespace.
After supporting CNP and ANP, with NetworkPolicyQueryFilter introduced, I want to keep the functionality of these two functions. `getNetworkPolicies` still return a list of NP that match the filter. `getNetworkPolicy` should accept a NetworkPolicyQueryFilter with attributes that can locate only one NP and return this NP. 
BTW I'm totally ok to remove function `getNetworkPolicy` since it can be covered by `getNetworkPolicies` now. If users want a single NP, they can pass a proper filter into `getNetworkPolicies` can get the first element of the returned list.